enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,320 @@
+
+### Alternative to the cluster capability to disable the ingress operator
+
+Due to the lack of understanding of how HyperShift uses the cluster version operator to manage the ingress operator, we thought that the capability would serve only the informative purpose for the other OpenShift components like the console operator.
+That brought the idea of giving up with the capability in favor of some other cluster resource (existing or not) which could fulfill the need of informing other OpenShift components.
+Potentially that could save us from the difficult design and implementation decisions which the cluster capability implies. [The hypershift chapter](#hypershift) explains in details why the capability should be preferred over any other API.
+
+### Add Route API to the ingress capability
+
+The Route API is a fundamental part of OpenShift, a lot of OpenShift components and user workloads rely on the routes. The impact of the Router API removal is difficult to anticipate and plan.
+Unlike the absence of the ingress operator which can be simulated without a dedicated capability, the removal of the route api needs the code changes in the [openshift-api-server](https://github.com/openshift/openshift-apiserver/tree/master/pkg/route).
+This puts the implementation of the RFE at risk. In the situation when the ingress is disabled the choice lies in between the following options:
+1. Keep the Route API and let the OpenShift components create routes which are not served by any ingress controller.
+2. Keep the Route API and let the OpenShift components use an alternative ingress which supports the Route API (e.g. [AWS Load Balancer Controller](/enhancements/ingress/aws-load-balancer-operator.md)).
+3. Remove the Route API and leave the OpenShift components adapt to the situation which implies the code changes to stop creating the routes and to migrate to a different ingress API (e.g. Kubernetes Ingress).
+
+The first option seems to be the easiest (for the implementation), the second is likely the most suitable for the long term needs (of the RFE this enhancement tries to address) and the third seems to be the most explicit and "honest" (with respect to the api users) but the most intrusive at the same time.
+The choice is not obvious as all the options have drawbacks.    
+However the second one seems to be the better match for the ROSA users which already want to use the AWS load balancers.
+Although the implementation of the second option requires the upstream work (AWS Load Balancer Controller doesn't support non Kubernetes APIs) and deserves a dedicated enhancement (and RFE).
+Since the first option doesn't exclude the second one it seems to be pragmatic to pick it leaving the room for the follow up.
```

Comment:
Minor copyedits.  
```suggestion
Although the implementation of the second option would require upstream work (AWS Load Balancer Controller doesn't support non-Kubernetes APIs, such as the Route API) and deserves a dedicated enhancement (and RFE),
since the first option doesn't preclude the second one, it seems to be pragmatic to pick the first option, leaving open the possibility of implementing the second option later on.
```